### PR TITLE
Update @ckeditor/ckeditor5-autosave to v29

### PR DESCRIPTION
**Pull request by fedemp**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/ckeditor/ckeditor5/releases/tag/v29.0.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/ckeditor__ckeditor5-autosave/ckeditor__ckeditor5-autosave-tests.ts [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
+const state: 'synchronized' | 'waiting' | 'saving' = 'saving';
+plugin.state === state;
 plugin.adapter.save(new MyEditor());
 
 const adapter: AutosaveAdapter = {
     save(editor: Editor) {
         return Promise.resolve(editor);
     },
 };
+adapter === adapter;
```
Comment (peterblazejewicz) [inline types/ckeditor__ckeditor5-autosave/ckeditor__ckeditor5-autosave-tests.ts:19]: what's the purpose of this? (and next one)

Comment (fedemp) [reply types/ckeditor__ckeditor5-autosave/ckeditor__ckeditor5-autosave-tests.ts]: It was to remove the warning about unused variables, but I finally found an use for that variable.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by fedemp** [comment]
@peterblazejewicz Sorry to ask for your review again, but I had to rebase due to recent upgrade to TS 4.3 in all ckeditor packages.

**Review by peterblazejewicz — APPROVED** [review]

**Comment by fedemp** [comment]
Ready to merge

**Pull request merged**